### ISSUE TITLE
Mounted files not available in `resources.Match`

### ISSUE THREAD

**Issue by XhmikosR** [gohugoio/hugo#6916]
### What version of Hugo are you using (`hugo version`)?

<pre>
$ hugo version
Hugo Static Site Generator v0.65.0/extended windows/amd64 BuildDate: unknown
</pre>

### Does this issue reproduce with the latest release?

Yes.

---

While trying to make https://github.com/twbs/bootstrap/pull/29283 work, I notice that the vendor JS files are not being concatenated.

https://github.com/twbs/bootstrap/blob/master-xmr-hugo-docs-vendor/site/layouts/partials/scripts.html#L11

Basically this is empty: `{{- $vendor := resources.Match "js/vendor/*.js" -}}`

---

Steps to reproduce:

1. `git clone https://github.com/twbs/bootstrap.git -b master-xmr-hugo-docs-vendor`
2. `npm i && npm run docs-serve` or just running `hugo` if you have it available

**Comment by bep** [comment]
You need to mount them inside `assets`.

**Comment by XhmikosR** [comment]
I do have them mounted but maybe I did it wrong?

https://github.com/twbs/bootstrap/blob/65fec9460ee9cb6f7856167fe36d8608e80dcee9/config.yml#L43

**Comment by bep** [comment]
>I do have them mounted but maybe I did it wrong?

Read my comment.

**Comment by XhmikosR** [comment]
Thanks, I finally got it to work :)

Now, only thing I wonder, is `resources.Match` supposed to not throw any error if the file(s) doesn't exist on purpose?

**Comment by bep** [comment]
Yes. It would be really hard to use from themes if that was not a case:

```
{ { $img := resources.GetMatch "logo.jpg" }}
{{ if not $img }}
// pick a default
{{ end }}
```

**Comment by XhmikosR** [comment]
Thanks for the help!

**Comment by bep** [comment]
A tip for you

```bash
{ { $img := resources.GetMatch "logo.jpg" }}
{{ if not $img }}
{{ errorf "fail the build" }}
{{ end }}
```
